### PR TITLE
async-ify more functions, and await completion

### PR DESCRIPTION
**Pull request by rhansen**
Where feasible I put the `await` at the end of the function to minimize the impact on latency.

My motivation for this change: Eliminate a race condition in tests I am writing.

### CODE REVIEW

**Comment by JohnMcLear** [comment]
My only concern here is if there are some scripts in bin/ that call some of these files directly that won't be using async/await?  I don't think they have test coverage.

**Comment by rhansen** [comment]
> My only concern here is if there are some scripts in bin/ that call some of these files directly that won't be using async/await? I don't think they have test coverage.

This PR does not cause any code that was previously synchronous to run asynchronously. All non-`async` callers of these newly-`async` functions were already continuing execution without waiting for the asynchronous code to complete—the only difference is they are now ignoring a Promise instead of ignoring an inaccessible background operation.

**Comment by rhansen** [comment]
Also: I used `git grep` to audit all callers of these functions to make sure they were either updated or already correct.

File: src/node/utils/ImportHtml.js [hunk 1]
```diff
@@ -103,6 +102,8 @@ exports.setPadHTML = function(pad, html)
   var theChangeset = builder.toString();
 
   apiLogger.debug('The changeset: ' + theChangeset);
-  pad.setText("\n");
-  pad.appendRevision(theChangeset);
+  await Promise.all([
+    pad.setText('\n'),
```
Comment (webzwo0i) [inline src/node/utils/ImportHtml.js:106]: Is it possible that this can lead to a race when `pad.appendRevision` completes before `pad.setText`?

Comment (rhansen) [reply src/node/utils/ImportHtml.js:106]: Looking at the code I don't think there's currently a race condition, but that could easily change if the implementation of `Pad.setText()` or `Pad.appendRevision()` is changed. Either way, this code is not very readable; waiting for `setText()` to complete before calling `appendRevision()` would be an improvement.

(Note that this code was problematic before—this PR didn't introduce the problem.)

**Pull request merged**